goal_src/jak2/levels/common/scene-actor.gc

```diff
@@ -5,5 +5,1193 @@
+  (case (-> obj task actor)
+    (((game-task-actor tess-alley))
+     (if (task-node-closed? (game-task-node ruins-tower-resolution))
+         (-> obj draw art-group data 3)
+         (-> obj draw art-group data 4)
+         )
+     )
+    (((game-task-actor tess-hiphog))
+     (cond
+       ((or (task-node-open? (game-task-node city-krew-collection-introduction))
+            (task-node-open? (game-task-node city-krew-collection-collection))
+            (task-node-open? (game-task-node forest-hunt-introduction))
+            (task-node-open? (game-task-node forest-hunt-resolution))
+            (task-node-open? (game-task-node city-errol-challenge-introduction))
+            (task-node-open? (game-task-node city-errol-challenge-race))
+            )
+        (-> obj draw art-group data 6)
+        )
+       ((and (task-node-closed? (game-task-node city-whack-pre-intro))
+             (not (task-node-closed? (game-task-node under-mech-resolution)))
+             )
+        (-> obj draw art-group data 5)
+        )
+       (else
+         (-> obj draw art-group data 3)
+         )
+       )
+     )
+    (else
+      (-> obj draw art-group data 3)
+      )
+    )
+  )
+
+(defmethod init-skeleton! tess-npc ((obj tess-npc))
+  "@see [[initialize-skeleton]]"
+  (initialize-skeleton
+    obj
+    (the-as skeleton-group (art-group-get-by-name *level* "skel-tess-highres" (the-as (pointer uint32) #f)))
+    (the-as pair 0)
+    )
+  (set! (-> obj draw light-index) (the-as uint 30))
+  0
+  (none)
+  )
+
+(deftype keira-npc (process-taskable)
+  ()
+  :heap-base #xa0
+  :method-count-assert 38
+  :size-assert         #x120
+  :flag-assert         #x2600a00120
+  )
+
+
+(defmethod get-art-elem keira-npc ((obj keira-npc))
+  "Checks various things such the current actor, task status, etc to determine the right art-group data to use
+       @returns the appropriate [[art-element]] for the given NPC"
+  (case (-> obj task actor)
+    (((game-task-actor keira-stadium))
+     (-> obj draw art-group data 3)
+     )
+    (else
+      (-> obj draw art-group data 3)
+      )
+    )
+  )
+
+(defmethod init-skeleton! keira-npc ((obj keira-npc))
+  "@see [[initialize-skeleton]]"
+  (initialize-skeleton
+    obj
+    (the-as skeleton-group (art-group-get-by-name *level* "skel-keira-highres" (the-as (pointer uint32) #f)))
+    (the-as pair 0)
+    )
+  (set! (-> obj draw light-index) (the-as uint 30))
+  0
+  (none)
+  )
+
+(deftype krew-npc (process-taskable)
+  ()
+  :heap-base #xa0
+  :method-count-assert 38
+  :size-assert         #x120
+  :flag-assert         #x2600a00120
+  )
+
+
+(defmethod get-art-elem krew-npc ((obj krew-npc))
+  "Checks various things such the current actor, task status, etc to determine the right art-group data to use
+       @returns the appropriate [[art-element]] for the given NPC"
+  (-> obj draw art-group data 4)
+  )
+
+(defmethod init-skeleton! krew-npc ((obj krew-npc))
```

Comment:
seems kinda confusing to have both `initialize-skeleton` and `init-skeleton!`. Maybe `init-skeleton!` could be `init-art` or something like that?